examples/runners/ash/src/main.rs

```diff
@@ -84,19 +84,51 @@ use std::{
     ffi::{CStr, CString},
     fs::File,
     os::raw::c_char,
+    path::PathBuf,
     sync::mpsc::{TryRecvError, TrySendError, sync_channel},
     thread,
 };
 
-use clap::Parser;
+use clap::{Parser, ValueEnum};
 
 use spirv_builder::{MetadataPrintout, SpirvBuilder};
 
 use shared::ShaderConstants;
 
+// This runner currently doesn't run the `compute` shader example.
+#[derive(Debug, PartialEq, Eq, Copy, Clone, ValueEnum)]
+pub enum RustGPUShader {
+    Simplest,
+    Sky,
+    Mouse,
+}
+
+impl RustGPUShader {
+    // The form with dashes, e.g. `sky-shader`.
+    fn crate_name(&self) -> &'static str {
+        match self {
+            RustGPUShader::Simplest => "simplest-shader",
+            RustGPUShader::Sky => "sky-shader",
+            RustGPUShader::Mouse => "mouse-shader",
+        }
+    }
+
+    // The form with underscores, e.g. `sky_shader`.
+    fn crate_ident(&self) -> &'static str {
```

Comment:
This can probably just be `self.crate_name().replace("-", "_")` (it's only used w/ `format!` AFAICT, so there's already an inherent string allocation cost anyway).

Reply:
Eh, but it has to return a `String`, which would make it different to `crate_name`, and I have a follow-up that will eliminate this anyway. So I'll leave it as is.